### PR TITLE
robots 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by peaceiris**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/r/robots.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      bin.install "robots"
+
+      if OS.mac?
+        lib.install "librobots.dylib"
+        require "macho"
+        MachO::Tools.add_rpath(bin/"robots", lib.to_s)
+      elsif OS.linux?
+        mv "librobots.so", "/usr/lib/librobots.so"
+      end
+    end
```
Comment (carlocab) [inline Formula/r/robots.rb:25]:
```suggestion
    system "cmake", "-S", ".", "-B", "c-build", "-DCMAKE_INSTALL_RPATH=#{rpath}", *std_cmake_args
    system "cmake", "--build", "c-build"
    system "cmake", "--install", "c-build"
```

1. Follow the pattern used in other formulae.
2. Don't build the tests.
3. Use `CMAKE_INSTALL_RPATH` so you don't need to adjust the RPATH manually.

**Review by carlocab — CHANGES_REQUESTED** [review]
Thanks for the PR. I have one suggestion below.

Also: this seems to download its own `googletest` and `abseil` at build-time (from `master`/`main`). We should use Homebrew `googletest` and `abseil` instead. See:

https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae
https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things

**Comment by peaceiris** [comment]
> https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things

I have opened the following issue to request to create a git tag.

- https://github.com/google/robotstxt/issues/68

File: Formula/r/robots.rb (hunk 1 above)
Comment (peaceiris) [reply Formula/r/robots.rb]: Fixed. Thank you!

**Comment by carlocab** [comment]
> > https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things
> 
> 
> 
> I have opened the following issue to request to create a git tag.
> 
> 
> 
> - https://github.com/google/robotstxt/issues/68
> 
> 

Thanks. There is also the issue of this downloading an unversioned `googletest` and an unversioned `abseil` at build-time. We should be able to use Homebrew dependencies instead.

Would you be able to submit a PR upstream that implements this?

**Comment by p-linnane** [comment]
Closing this until upstream makes a tag. Feel free to reopen once that happens.

**Pull request closed without merging**